File: dotnet/DesignPatterns/StructuralDesignPatterns/ProxyDesignPattern/ProxyDesignPattern.DynamicProxyExample/Interfaces.cs
namespace ProxyDesignPattern.DynamicProxyExample;

/// <summary>
/// Sample interface for a calculator service.
/// </summary>
public interface ICalculator
{
    int Add(int a, int b);
    int Subtract(int a, int b);
    int Multiply(int a, int b);
    double Divide(int a, int b);
}

/// <summary>
/// Sample interface for a greeting service.
/// </summary>
public interface IGreetingService
{
    string Greet(string name);
    string Farewell(string name);
}

File: nodejs/DesignPatterns/StructuralDesignPatterns/ProxyDesignPattern/ProxyDesignPattern.DynamicProxyExample/src/Interfaces.ts
/**
 * Sample interface for a calculator service.
 */
export interface ICalculator {
  add(a: number, b: number): number;
  subtract(a: number, b: number): number;
  multiply(a: number, b: number): number;
  divide(a: number, b: number): number;
}

/**
 * Sample interface for a greeting service.
 */
export interface IGreetingService {
  greet(name: string): string;
  farewell(name: string): string;
}
